### PR TITLE
🎉 Source Zendesk Sunshine: support oauth

### LINKED ISSUE #7708: Source Zendesk Sunshine: support oauth
## Tell us about the problem you're trying to solve
With the release of Airbyte Cloud, we need to start supporting Oauth for this connector, since it's the recommended way of authenticating users into a SaaS application.

If this connector doesn't support oauth already (i.e: doesn't accept a client_id and client_secret) then we need to update its spec to accept those parameters. There are two ways to do this:

If the connector already supports some auth mechanism like api_key, I suggest that this be a oneof nested inside a top-level field called "authentication":

{ authentication: { type: object oneOf: [ // api key, // oauth ] } }

If the connector only supports webflow oauth, then no changes are needed to the properties format and we will only need to add annotations.

See the connector spec reference in the docs for reference on how a oneof can be implemented.

This should be done in a backwards compatible manner i.e: users currently supplying authentication info in the config's top-level should not be impacted by this change.
##Acceptance criteria
The connector supports oauth webflow authentication with client_id/client_secret in a backwards compatible manner
Oauth properties are annotated properly. See this PR for an example

### PR DESCRIPTION
**Pull request by annalvova05**
## What
#7708 Add oauth2.0 flow part for Zendesk Sunshine.
zd-sunshine-1
zd-sunshine-2

## How
Change `spec.json` and implement new Auth method in `source.py`

## Recommended reading order
1. `spec.json`
2. `source.py`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/source.py [hunk 1]
```diff
@@ -23,11 +24,37 @@ def __init__(self, auth: Tuple[str, str], auth_method: str = "Basic", **kwargs):
...
+
+    @staticmethod
+    def get_auth(config: Mapping[str, Any]) -> AuthBase:
+        credentials = config.get("credentials", {})
+        auth_method = credentials.get("auth_method")
+
+        if auth_method == "api_token" or not credentials:
+            api_token = credentials.get("api_token") or config.get("api_token")
+            if not api_token:
+                raise Exception("No api_token in creds")
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/source.py:38]: Please, make this log message more informative.

File: airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/source.py [hunk 2]
```diff
@@ -23,11 +24,37 @@ def __init__(self, auth: Tuple[str, str], auth_method: str = "Basic", **kwargs):
...
+            email = credentials.get("email") or config.get("email")
+            if not email:
+                raise Exception("No email in creds")
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/source.py:41]: Please, make this log message more informative.

File: airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/streams.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 
 class SunshineStream(HttpStream, ABC):
-    primary_key = "id"
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/streams.py:17]: Why do we changed this to `None` ?

**Review by bazarnov — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/streams.py [hunk 4]
```diff
@@ -77,6 +77,8 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
 
 
 class ObjectTypes(SunshineStream):
+    primary_key = "key"
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/streams.py:80]: Is it possible to have the `primary_key` property mainly in `SunshineStream` and re-assign only if any other stream has something else but `key` ?

**Review by bazarnov — APPROVED** [review]
LGTM, with a little question regards having `primary_key` inside of parent class. Check it, please.

File: airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/streams.py (hunk 4 above)
Comment (annalvova05) [reply airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/streams.py:80]: There is set the `primary_key = "id"` in the `SunshineStream`. And if any other stream has another `primary_key`, it is overwritten.

**Comment by annalvova05** [comment]
Java tests was passed
image

**Review by antixar — APPROVED** [review]
LGTM! Please publish this

**Pull request merged**